hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/ssl/ReloadingX509KeyManager.java

```diff
@@ -67,27 +66,30 @@ public class ReloadingX509KeyManager extends X509ExtendedKeyManager {
    * materials are changed.
    */
   private PrivateKey currentPrivateKey;
-  private List<String> currentCertIdsList = new ArrayList<>();
-  private String alias;
+  private List<X509Certificate> currentTrustChain;
+  private final String alias;
 
   /**
    * Construct a <code>Reloading509KeystoreManager</code>.
    *
-   * @param type type of keystore file, typically 'jks'.
-   * @param caClient client to get the private key and certificate materials.
+   * @param type          type of keystore file, typically 'jks'.
+   * @param componentName the name of the component for which the keys are created.
+   * @param privateKey    private key for this key manager.
+   * @param trustChain    list of the trusted certificates.
    * @throws IOException
    * @throws GeneralSecurityException
    */
-  public ReloadingX509KeyManager(String type, CertificateClient caClient)
+  public ReloadingX509KeyManager(String type, String componentName, PrivateKey privateKey,
+      List<X509Certificate> trustChain)
       throws GeneralSecurityException, IOException {
     this.type = type;
     keyManagerRef = new AtomicReference<>();
-    keyManagerRef.set(loadKeyManager(caClient));
+    this.alias = componentName + "_key";
```

Comment:
nit: can we move this up by one line before the instantiation of keyManagerRef?

Reply:
Sure thing